tutorpicasso/commands/run_extra_commands.py

```diff
@@ -0,0 +1,154 @@
+import re
+import subprocess
+from itertools import chain
+
+from typing import Any, List
+
+import click
+from tutor import config as tutor_config
+
+COMMAND_CHAINING_OPERATORS = ["&&", "&", "||", "|", ";"]
+
+
+@click.command(name="run-extra-commands", help="Run tutor commands")
+def run_extra_commands() -> None:
+    """
+    This command runs tutor commands defined in PICASSO_EXTRA_COMMANDS
+    """
+    context = click.get_current_context().obj
+    tutor_conf = tutor_config.load(context.root)
+
+    picasso_extra_commands: Any = tutor_conf.get("PICASSO_EXTRA_COMMANDS", None)
+
+    if not picasso_extra_commands:
+        return
+
+    error_message = validate_commands(picasso_extra_commands)
+    if error_message:
+        raise click.ClickException(error_message)
+
+    list(map(run_command, picasso_extra_commands))
+
+
+def validate_commands(commands: Any) -> str:
+    """
+    Takes all the extra commands sent through config.yml and verifies that
+    all the commands are correct before executing them
+
+    Args:
+        commands (list[str] | None): The commands sent through PICASSO_EXTRA_COMMANDS in config.yml
+    """
+    splitted_commands = [
+        split_string(command, COMMAND_CHAINING_OPERATORS) for command in commands
+    ]
+    flat_commands_list: chain[str] = chain.from_iterable(splitted_commands)
+
+    invalid_commands = []
+    misspelled_commands = []
+    for command in flat_commands_list:
+        if "tutor" in command.lower():
+            continue
+
+        if find_tutor_misspelled(command):
+            misspelled_commands.append(command)
+        else:
+            invalid_commands.append(command)
+
+        error_message = ""
+
+        if invalid_commands:
+            error_message += (
+                f"Found some issues with the commands:\n\n"
+                f"=> Invalid commands: {', '.join(invalid_commands)}\n"
+            )
+
+        if misspelled_commands:
+            error_message += (
+                f"=> Misspelled commands: {', '.join(misspelled_commands)}\n"
+            )
+
+        if error_message:
+            error_message += (
+                "Take a look at the official Tutor commands: "
+                "https://docs.tutor.edly.io/reference/cli/index.html"
+            )
+            return error_message
```

Comment:
question: if multiple commands have errors, will this only return the first it encounters? what if returns them all https://github.com/eduNEXT/tutor-contrib-picasso/commit/87e85030e8fc478e05f8d71e424f9fe10b0b111b?

Reply:
Thanks for this. I applied https://github.com/eduNEXT/tutor-contrib-picasso/commit/87e85030e8fc478e05f8d71e424f9fe10b0b111b and applied a little fix. You can check and verify if everything is okay.